manage.py

```diff
@@ -55,6 +54,20 @@ def __init__(self, args: List[str]):
             os.environ.setdefault("DEBUG", "true")
             print("Starting in debug mode.")
 
+    @staticmethod
+    def parse_db_url(db_url: str) -> SplitResult:
+        """Validate and split the given databse url."""
+        db_url_parts = urlsplit(db_url)
+        if not all((
+            db_url_parts.hostname,
+            db_url_parts.port,
+            db_url_parts.username,
+            db_url_parts.password,
+            db_url_parts.path
+        )):
+            raise OSError("Valid DATABASE_URL environment variable not found.")
```

Comment:
This error message is inaccurate and the type of exception is inappropriate. It should be a `ValueError` that states the URL is invalid. A `KeyError` will already be raised by `os.environ["DATABASE_URL"]` if the variable doesn't exist.

Reply:
Oh it does state its invalid, I missed that. Well, just change OSError then I guess.

Reply:
I think it's a bit confusing since it says "not found" at the end.

Reply:
Yea agreed, I copy-pasted it from the previous validation without reading it. Updated it in https://github.com/python-discord/site/pull/583/commits/4a0207fc198feab280065772142dbaffe25ffb4f